src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -66,6 +69,29 @@ export class PtyService extends Disposable implements IPtyService {
 		}));
 	}
 
+	private _pendingDetachInstanceRequests: Map<number, (resolved: IProcessDetails | undefined) => void> = new Map();
+	async requestDetachInstance(workspaceId: string, instanceId: number): Promise<IProcessDetails | undefined> {
+		return new Promise<IProcessDetails | undefined>(resolve => {
+			const requestId = ++lastResolvedInstanceRequestId;
+			this._pendingDetachInstanceRequests.set(requestId, resolve);
+			this._onDidRequestDetach.fire({ requestId, workspaceId, instanceId });
+		});
+	}
```

Comment:
An improvement we could make here would be to have a max timeout before the promise is rejected such that there isn't the unresolved promise in the target window forever: https://github.com/microsoft/vscode/issues/128786